dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -145,7 +124,9 @@ class Recurrent[T : ClassTag] (
     var i = times
     while (i >= 1) {
       currentGradOutput(inputDim) = gradOutput.select(timeDim, i)
-      modules(i - 1).accGradParameters(inputs(i), currentGradOutput, scale)
+      val _input = T(if (i > 1) modules(i - 2).output.toTable(hidDim)
+      else hidden, input.select(timeDim, i))
```

Comment:
This can be something like:

    var _hidden = hidden
    while (i >= 1) {
      ...
      val _input = T(_hidden, input.select(timeDim, i))
      ...
      _hidden = modules(i - 1).output.toTable(hidDim)
    }